### PR TITLE
fix #6512 Cyclemap via Thunderforest with API key

### LINKED ISSUE #6512: OSM:Cyclemap requires API key
##### Detailed steps causing the problem:
- Use live map with OSM:Bikemap (or whatever it is called in english)

##### Actual behavior after performing these steps:
Tiles have a text overlay API key required

##### Expected behavior after performing these steps:
No warning like this

##### Version of c:geo used:
2017.03.29

##### Is the problem reproducible for you?
Yes

### PR DESCRIPTION
**Pull request by pstorch**
For the old map I also had to update https://github.com/cgeo/mapsforge, I'll send a PR there, too.

### CODE REVIEW

File: main/src/cgeo/geocaching/maps/mapsforge/v6/layers/ThunderforestMap.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    public ThunderforestMap(final String[] hostNames, final int port) {
+        super(hostNames, port);
+    }
+
+    public int getParallelRequestsLimit() {
+        return 8;
+    }
+
+    public URL getTileUrl(final Tile tile) throws MalformedURLException {
+        return new URL("https", this.getHostName(), this.port, "/cycle/" + tile.zoomLevel + '/' + tile.tileX + '/' + tile.tileY + ".png?apikey=d6866308a6e24783822338c9a49ceb6c");
```
Comment (Bananeweizen) [inline main/src/cgeo/geocaching/maps/mapsforge/v6/layers/ThunderforestMap.java]: I suggest to remove the key similarly to all other keys we use:
* create a new resource string in keys.xml (and in its template version), including a comment where to register for a new key
* use the resource string in the code
* have @mucek4 put the actual value into the file on the CI server

Comment (kumy) [inline main/src/cgeo/geocaching/maps/mapsforge/v6/layers/ThunderforestMap.java]: if @mucek4 is unavailable I could do it. :)

Comment (pstorch) [inline main/src/cgeo/geocaching/maps/mapsforge/v6/layers/ThunderforestMap.java]: I can externalize the key in the mapsforge V6 version, but not in the V3. This is handled in the other repo and I doubt I can access the keys from there.

**Comment by pstorch** [comment]
@Bananeweizen, @kumy, @mucek4 the key is externalized and CI should replace `@thunderforest.api.key@` in the `templates/keys.xml` file.

**Comment by kumy** [comment]
Key added to `private.properties` on server.

**Comment by kumy** [comment]
@samueltardieu it may also be good to add the key on your executor?

**Comment by samueltardieu** [comment]
@kumy Is that needed? There doesn't seem to be any test using this map, and as long as we don't share other keys and signatures (which we should IMO, as artifacts would be usable by everyone, including on pull requests), this seems useless to me.

**Comment by kumy** [comment]
I was thinking about unit tests, if there is none, then I agree, it's not necessary.

**Pull request merged**